loopy/codegen/loop.py

```diff
@@ -472,14 +472,16 @@ def generate_sequential_loop_dim_code(codegen_state, sched_index):
 
             from loopy.isl_helpers import simplify_pw_aff
 
-            result.append(
-                inner.with_new_ast(
-                    codegen_state,
-                    astb.emit_sequential_loop(
+            loop = astb.emit_sequential_loop(
                         codegen_state, loop_iname, kernel.index_dtype,
                         pw_aff_to_expr(simplify_pw_aff(lbound, kernel.assumptions)),
                         pw_aff_to_expr(simplify_pw_aff(ubound, kernel.assumptions)),
-                        inner_ast)))
+                        inner_ast)
+
+            from cgen import Block
+            loop = Block(list(pragmas) + [loop])
```

Comment:
I feel like `Block` is the wrong thing here. Any ideas?

Reply:
`Collection`. I've pushed that change.